File: Ede.Uofx.Customize.Web/ClientApp/src/app/model/northwind.model.ts

/** 客戶 model */
export interface CustomerModel {
  /** 客戶id */
  customerId: string;
  /** 公司名稱 */
  companyName: string;
  /** 聯絡人 */
  contactName: string;
  /** 手機 */
  phone: string;
}

/** 商品 model */
export interface ProductModel {
  /** 商品id */
  productID: number;
  /** 商品名稱 */
  productName: string;
  /** 單價 */
  unitPrice: number;
  /** 庫存 */
  unitsInStock: number;
  /** 數量 */
  quantity: number;
}

File: Ede.Uofx.Customize.Web/Models/NorthWindModel.cs
namespace Ede.Uofx.Customize.Web.Models
{
    public class CustomerModel
    {
        public string CustomerID { get; set; }
        public string CompanyName { get; set; }
        public string? ContactName { get; set; }
        public string? Phone { get; set; }
    }

    public class CategoryModel
    {
        public decimal CategoryID { get; set; }
        public string CategoryName { get; set; }
    }

    public class ProductGetByCategoryExternalModel
    {
        public ExternalModel? CategoryID { get; set; }
    }

    public class ProductExternalModel
    {
        public ExternalModel? ProductID { get; set; }
    }

    public class ProductModel
    {
        public decimal ProductID { get; set; }
        public string? ProductName { get; set; }
        public decimal Quantity { get; set; }
        public decimal? UnitPrice { get; set; }
        public decimal? UnitsInStock { get; set; }
    }

    public class UpdateProductExternalModel
    {
        public ExternalModel ProductID { get; set; }
        public decimal Quantity { get; set; }
    }

    public class UpdateProductModel
    {
        public decimal ProductID { get; set; }
        public decimal Quantity { get; set; }
    }

    public class OrderModel
    {
        public decimal? OrderID { get; set; }
        public DateTimeOffset OrderDate { get; set; }
        public List<OrderDetailModel> OrderDetails { get; set; }
    }

    public class OrderDetailModel
    {
        public decimal? OrderID { get; set; }
        public decimal ProductID { get; set; }
        public decimal Quantity { get; set; }
        public decimal UnitPrice { get; set; }
    }

    public class ExternalModel
    {
        public string Selected { get; set; }
        public string FillText { get; set; }
        public bool IsOther { get; set; }
    }
}
